src/screens/Signup/StepHandle.tsx

```diff
@@ -103,115 +124,212 @@ export function StepHandle() {
-              </View>
-            ) : undefined}
-            {validCheck.hyphenStartOrEnd ? (
-              <View style={[a.w_full, a.flex_row, a.align_center, a.gap_sm]}>
-                <IsValidIcon valid={validCheck.handleChars} />
-                <Text style={[a.text_md, a.flex_1]}>
-                  <Trans>Only contains letters, numbers, and hyphens</Trans>
-                </Text>
-              </View>
-            ) : (
-              <View style={[a.w_full, a.flex_row, a.align_center, a.gap_sm]}>
-                <IsValidIcon valid={validCheck.hyphenStartOrEnd} />
-                <Text style={[a.text_md, a.flex_1]}>
-                  <Trans>Doesn't begin or end with a hyphen</Trans>
-                </Text>
-              </View>
+        <LayoutAnimationConfig skipEntering skipExiting>
+          <View style={[a.gap_xs]}>
+            {state.error && (
+              <Requirement>
+                <RequirementText>{state.error}</RequirementText>
+              </Requirement>
             )}
-            <View style={[a.w_full, a.flex_row, a.align_center, a.gap_sm]}>
-              <IsValidIcon
-                valid={validCheck.frontLength && validCheck.totalLength}
-              />
-              {!validCheck.totalLength ||
-              draftValue.length > MAX_SERVICE_HANDLE_LENGTH ? (
-                <Text style={[a.text_md, a.flex_1]}>
+            {queryEnabled &&
+              isHandleAvailable &&
+              !isHandleAvailable.available &&
+              validCheck.overall && (
+                <>
+                  <Requirement>
+                    <RequirementText>
+                      <Trans>
+                        {createFullHandle(draftValue, state.userDomain)} is not
+                        available
+                      </Trans>
+                    </RequirementText>
+                  </Requirement>
+                  {isHandleAvailable.suggestions &&
+                    isHandleAvailable.suggestions.length > 0 && (
+                      <HandleSuggestions
+                        suggestions={isHandleAvailable.suggestions}
+                        onSelect={suggestion => {
+                          setDraftValue(
+                            suggestion.handle.slice(
+                              0,
+                              state.userDomain.length * -1,
+                            ),
+                          )
+                          logger.metric('signup:handleSuggestionSelected', {
+                            method: suggestion.method,
+                          })
+                        }}
+                      />
+                    )}
+                </>
+              )}
+            {(!validCheck.handleChars || !validCheck.hyphenStartOrEnd) && (
+              <Requirement>
+                {!validCheck.hyphenStartOrEnd ? (
+                  <RequirementText>
+                    <Trans>Doesn't begin or end with a hyphen</Trans>
```

Comment:
```suggestion
                    <Trans>Username cannot begin or end with a hyphen</Trans>
```
Given the different context this error string now appears in, I suggest rewording it slightly so that it reads a bit better